src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4182,6 +4182,147 @@ def __handle_zfs_set_property_error(self, e, properties_definitions):
             raise CallError(pjob.error)
         return data
 
+    @accepts(
+        Str('id', required=True),
+        Str('type', enum=['USER', 'GROUP']),
+        Ref('query-filters'),
+        Ref('query-options'),
+    )
+    @item_method
+    async def get_userquota(self, id, type, filters, options):
```

Comment:
Should we perhaps rename this to `get_quota` as it can retrieve quotas for groups as well

Reply:
I'm someone unsure of what to do here. `get_quota` might be confusing because we have dataset quotas as well as user and group quotas that apply to datasets. In one iteration I had `get_user_or_group_quota` if you think that's less confusing.

Reply:
I like `get_zfs_user_or_group_quota`.

Reply:
Went with maximal length